dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ConfigurationRestRepository.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import javax.annotation.Resource;
+
+import org.dspace.app.rest.exception.RepositoryMethodNotImplementedException;
+import org.dspace.app.rest.model.PropertyRest;
+import org.dspace.core.Context;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is the repository responsible of exposing configuration properties
+ */
+@Component(PropertyRest.CATEGORY + "." + PropertyRest.NAME)
+public class ConfigurationRestRepository extends DSpaceRestRepository<PropertyRest, String> {
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Resource(name = "exposedConfigurationProperties")
+    private ArrayList<String> exposedProperties;
+
+    /**
+     * Gets the value of a configuration property if it is exposed via REST
+     *
+     * Example:
+     * <pre>
+     * {@code
+     * curl http://<dspace.server.url>/api/config/properties/google.analytics.key
+     *  -XGET \
+     *  -H 'Authorization: Bearer eyJhbGciOiJI...'
+     * }
+     * </pre>
+     *
+     * @param property
+     * @return
+     */
+    @Override
+    public PropertyRest findOne(Context context, String property) {
+        if (!exposedProperties.contains(property) || configurationService.getArrayProperty(property).length == 0) {
+            throw new ResourceNotFoundException("No such configuration property: " + property);
+        }
+
+        String[] propertyValues = configurationService.getArrayProperty(property);
+
+        PropertyRest propertyRest = new PropertyRest();
+        propertyRest.setName(property);
+        propertyRest.setValues(Arrays.asList(propertyValues));
+
+        return propertyRest;
+    }
+
+    @Override
+    public Page<PropertyRest> findAll(Context context, Pageable pageable) {
+        throw new RepositoryMethodNotImplementedException("No implementation found; Method not allowed", "");
```

Comment:
As I was reviewing this code, I've just now realized that I'm not sure whether we want a 405 response on the main endpoint (which I know we decided on in the contract).  If this returns a 405, that is more secure, but how will the Client ever know which properties are available?  Do we just need to assume the Client is hardcoded to know what properties are server side properties?

This isn't a question we *have* to resolve in this particular PR, it could be resolved in a followup PR.  But, we might be setting up a problem similar to the one described in [DS-4495](https://jira.lyrasis.org/browse/DS-4495) by completely hiding the list of configurations from the client side.  Maybe @artlowel should look at this PRs approach to see if it will work for the Angular UI?

Reply:
I don't really see a way past hardcoding the property names we need for a specific feature in the client. E.g. the client side code that needs to configure the google analytics ID needs to know some name for the property it needs from the server. It can't just iterate over the list of all properties until it finds something that looks like a google analytics ID.

As long as the property name is the only part of the url we need to hardcode we'll be fine. It's not that different from filling in the UUID in the findById for an item for example. We also don't get the complete HAL link from the server in those cases.

The only use case to have the findAll endpoint available I can think of right now would be an overview of the server config for admins, which could be useful, but isn't planned right now as far as I know.

Reply:
@artlowel : Good points, and you are right. I'll flag this comment as resolved, as we can just keep the 405 response here for now.